modules/serverless-rn-1-17-0.adoc

```diff
@@ -0,0 +1,60 @@
+* The `kn func` CLI plug-in now uses `func` 0.17.0.
+* In the upcoming {ServerlessProductName} 1.19.0 release, the URL scheme of external routes will default to HTTPS for enhanced security.
++
+If you do not want this change to apply for your workloads, you can override the default setting before upgrading to 1.19.0, by adding the following YAML to your `KnativeServing` custom resource definition (CRD):
++
+[source,yaml]
+----
+...
+spec:
+  config:
+    network:
+      defaultExternalScheme: "http"
+...
+----
+
+* mTLS functionality is now Generally Available (GA).
+
+* TypeScript templates are now available when you create a function using `kn func`.
+
+* Changes to API versions in Knative Eventing 0.23.0:
+
+** Version `v1beta1` has been removed from the `eventing.knative.dev`, `messaging.knative.dev` and `flows.knative.dev` APIs. You must ensure that any existing or new resources in your deployment that are created by using these APIs use version `v1`.
+
+** The `v1alpha1` version of the `KafkaChannel` API, which was deprecated in {ServerlessProductName} version 1.14.0, has been removed. If the `ChannelTemplateSpec` of your config maps contain references to this older version, you must update this part of the spec to use the correct API version.
+
+** The deprecated `v1alpha1`, `v1alpha2`, and `v1beta1` versions of the `APIServerSource` and `SinkBinding` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** The deprecated `v1alpha2` and `v1beta1` versions of the `ContainerSource` and `PingSource` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** Version `v1beta2` of the `PingSource` API has been deprecated. References to older versions of this API should be updated to `v1` to prepare your deployment for removal of the deprecated version.
+
+[id="fixed-issues-1-17-0_{context}"]
+== Fixed issues
+
+[id="known-issues-1-17-0_{context}"]
+== Known issues
+
+* If you try to use an older version of the Knative `kn` CLI with a newer {ServerlessProductName} release, the API is not found and an error occurs.
++
+For example, if you use the 1.16.0 release of the `kn` CLI, which uses version 0.22.0, with the 1.17.0 {ServerlessProductName} release, which uses the 0.23.0 versions of the Knative Serving and Knative Eventing APIs, the CLI does not work because it continues to look for the outdated 0.22.0 API versions.
++
+Ensure that you are using the latest `kn` CLI version for your {ServerlessProductName} release to avoid issues.
+
+* If you create a new subscription for a Kafka channel, there might be a delay in the Kafka channel data plane becoming ready to dispatch messages after the newly created subscription reports a ready status. As a result, messages that are sent during the time which the data plane is not reporting a ready status might not be delivered to the subscriber.
++
+The following steps are recommended to mitigate this issue:
```

Comment:
Id also add some steps on how to recover if we fail to follow mitigate instructions:

```asciidoc
+Recovery options in case of failure to mitigate this problem:
* KafkaSource/KafkaChannel do not delete any data on underlaying Kafka topic.
* If it's possible, deduce the missing data period. If your system tolerates event duplication, you can choose this period with some overlap
* To restore missing parts of the data in downstream of KafkaChannel/KafkaSource you should modify the offsets on appopriate Kafka topic consumer-group
+
[source,bash]
----
$ oc exec -it my-cluster-kafka-0 -n kafka -- /bin/sh
sh-4.4$ bin/kafka-consumer-groups.sh --bootstrap-server <kafkahost:port> \
  --group <group_id> \
  --topic <topic_name> \
  --reset-offsets \
  --to-datetime <XXXX-XX-XXTXX:XX:XX.XXX> \
  --execute
----
+
* Delete pods for KafkaSource, located in the same namespace as KafkaSource. Eventing will recreate them with proper offsets.
* Delete pods for Kafka channel `kafka-ch-controller*` and `kafka-ch-dispatcher*`
```

This makes sense @devguyio ?

![Consumer Group offsets](https://user-images.githubusercontent.com/540893/133164136-6c7abc65-8f06-4ab5-ada2-e073945698cf.png)

Reply:
:cry: Tried that, but:

```bash
sh-4.4$ bin/kafka-consumer-groups.sh --bootstrap-server my-cluster-kafka-bootstrap.kafka.svc:9092 --group kafka.demo.example.b758f9e2-7e7c-490a-92a9-9d746730fb7a --topic knative-messaging-kafka.demo.example --reset-offsets --to-offset 0 --execute

Error: Assignments can only be reset if the group 'kafka.demo.example.b758f9e2-7e7c-490a-92a9-9d746730fb7a' is inactive, but the current state is Stable.

GROUP                          TOPIC                          PARTITION  NEW-OFFSET 
```

Reply:
That is not always a mitigation and I'd refrain from suggesting it because events replay needs idempotent subscribers. At most we can just suggest "consider resetting the offsets if you have idempotent subscribers" and that's it. I can imagine folks using Kafka in production can very well do a simple offset reset.

Reply:
They can't do it without stopping KnativeKafka elements. They makes the consumer group being in Stable state, and you can't modify offset while it is like that:

```
sh-4.4$ bin/kafka-consumer-groups.sh --bootstrap-server my-cluster-kafka-bootstrap.kafka.svc:9092 --group kafka.demo.example.b758f9e2-7e7c-490a-92a9-9d746730fb7a --topic knative-messaging-kafka.demo.example --reset-offsets --to-offset 0 --execute

Error: Assignments can only be reset if the group 'kafka.demo.example.b758f9e2-7e7c-490a-92a9-9d746730fb7a' is inactive, but the current state is Stable.

GROUP                          TOPIC                          PARTITION  NEW-OFFSET
```

Also, I wrote: 
>_If your system tolerates event duplication, you can choose this period with some overlap_

Reply:
> If your system tolerates event duplication, you can choose this period with some overlap
I'm sorry I missed that . 

I think we can tinker on this for long, let's try to stick to generic advises is in the [updated suggestion](https://github.com/openshift/openshift-docs/pull/35500#discussion_r707636599) . 

Since it's not possible to give a specific solution that fits everybody, let's try to stay generic:
1. manual verification when possible
2. automation : event acknowledgment or some other safety measure. 

It's not an ideal situation so I don't think we'll cover all different scenarios here.

Reply:
>I think we can tinker on this for long, let's try to stick to generic advises as in the updated suggestion .

+1.

I think we're getting way too far off into the weeds and basically starting to scope a troubleshooting section at this point for something that is a temporary known issue.
The point of calling it out as a known issue is that users should expect there to be an issue / the feature might not work right now - we don't need to provide a complete workaround for every single user complete with troubleshooting if the solution doesn't work, IMO.
This should not be expected as part of the scope of a release note, and we need to break the habit of trying to completely document our way around usability issues.